### ISSUE TITLE
Header links are not aligned with site name

### ISSUE THREAD

**Issue by quicksketch** [backdrop-contrib/basis#39]
In the mockup, header links appear to be centered with the site name text:

![3qoqxrp5oj](https://cloud.githubusercontent.com/assets/100206/18068629/238f97d6-6df7-11e6-86d5-26887dd3b696.png)

But the current theme seems to align these items at the top(?) rather than middle or baseline.

![selection_167](https://cloud.githubusercontent.com/assets/100206/18068698/91ee240e-6df7-11e6-97a5-96009ecc2614.png)

**Comment by quicksketch** [comment]
Oh, I think I see what might be happening... The "header menu" (the menu configured in the block) is shown in large text, with the "main menu" (or "Primary Navigation") block being shown right next to it. I think I remember @wesruv mentioning this problem on one of the design meeting hangouts.

If we follow the current paradigms for what links are placed in which menus, then we'd have to do something like this:

![8adtrtdmce](https://cloud.githubusercontent.com/assets/100206/18069822/6a187324-6dfe-11e6-8d86-266280fd5642.png)

Which has some challenges because the header block contents (the site name, logo #10, and header menu) are on either side of the main menu block, but in the DOM they are all prior to the main menu.

Lots of possible problems could happen here. What happens when there are too many menu items in either menu? How can we align them with each other when the DOM positioning is unexpected?

I'll see what sort of options we have here.

**Comment by quicksketch** [comment]
The current implementation is all based in `layout.css` and uses flexbox to align items. As I don't think flexbox allows you to rearrange items that are nested as such, I think the only way to do without modifying the markup is to:
- Remove flexbox implementation of the header block, use floats instead when needed.
- Use `display: inline-block` on the elements for site branding (logo and site name), the header menu, and the main menu.
- Float the site name left and the header menu right. These can be aligned with each other vertically even though they are floated in opposite directions.
- Remove any "clearfix" on the header block, which will allow the main menu block to drift up into the header block.
- As the header menu is also an `inline-block` element, it can be vertically aligned with the site branding.

I think this will work, but I haven't tried it out yet.

There are a bunch of `@todo` items stating that it would be preferrable to remove the concept of header block menus entirely, but I doubt that we'd be able to do that in a 1.x cycle. As such are options are either 1) use CSS to reposition existing markup (maybe possible as described above) or 2) Use theme/preprocessing functions to adjust the markup. I tried adjusting the markup when these elements were next to each other, but it was getting really hacky really fast.  So if it's possible to do with CSS, that would be a lot cleaner and less hacky.

**Comment by quicksketch** [comment]
> Float the site name left and the header menu right. These can be aligned with each other vertically even though they are floated in opposite directions.

Doh, this doesn't work. Items loose their vertical alignment when floated. Figures. This idea might still work if we set a fixed (minimum) height on the branding and header menu. More tinkering required.

**Comment by quicksketch** [comment]
PR filed at https://github.com/backdrop-contrib/basis/pull/47. This needs some review for sure.
- I moved the header CSS out of `layout.css` and into `components/header.css`, but while 80% of that file is the "header block", there is definitely some related to "header region" (and the blocks within it). Not sure about how kosher this is.
- Vertically aligning the items in the header mostly works. The header now has a fixed minimum height. This should work well up to about 6 menu items, at which point the main menu links will start running over the user account links. Flexbox is used for vertical centering, but it's use is pretty optional. Without it, elements are simply vertically aligned instead, which still looks okay for browsers that don't support it.
- I added the heading for the header menu that says `Hi [username]!`. Not 100% about this implementation (or the use of an H2 for the tag there). See the changes in template.php.

Here's a screenshot of everything in place:
![2016-08-29 20 37 24 942](https://cloud.githubusercontent.com/assets/100206/18075310/76c738ce-6e28-11e6-9347-657a445df88a.gif)
